lib/http.c

```diff
@@ -1509,6 +1509,18 @@ static bool use_http_1_1plus(const struct Curl_easy *data,
           (data->set.httpversion >= CURL_HTTP_VERSION_1_1));
 }
 
+static const char *get_http_string(const struct Curl_easy *data,
+                                   const struct connectdata *conn)
+{
+  if(data->set.httpversion >= CURL_HTTP_VERSION_2_0)
```

Comment:
This doesn't take `CURL_HTTP_VERSION_2TLS` into account. It makes the selection a bit more complicated...

Also, this is what's *requested*, not what is necessarily negotiated. Try `https://www.sourceforge.net/` - which right now is a HTTPS server that doesn't do h2, even when curl uses "2TLS" with it. With this condition it says "GET / HTTP/2", but it is a lie!